tests/binary-with-tool/Main.hs

```diff
@@ -0,0 +1,6 @@
+{-# OPTIONS_GHC -F -pgmF bazel-out/host/bin/tests/binary-with-tool/cat #-}
```

Comment:
Perfectly fine for this test-case, though I wonder if there's a way to avoid hard-coding that path. Maybe if we'd expand `$(location ...)` in `compiler_flags`?

Reply:
Well, now you're asking for another feature. ;) Which happens to be the subject of the next PR. Though I'm not sure it will help this particular case. This path is at least stable across architectures, and seems to arise because the build is happening in one configuration while the tool was built in another. It appears that none of it should be considered Bazel private, but rather a stable Bazel convention for how to name things.